packages/inter-protocol/src/proposals/upgrade-vaults.js

```diff
@@ -0,0 +1,202 @@
+      // @ts-expect-error It's an amount.
+      while (AmountMath.isEmpty(value.current.DebtLimit.value)) {
+        ({ value, updateCount } = await notifier.getUpdateSince(updateCount));
+        trace(`debtLimit was empty, retried`, value.current.DebtLimit.value);
+      }
+      trace(kwd, 'params at', updateCount, 'are', value.current);
+      params[kwd] = harden({
+        brand: collateralBrand,
+        debtLimit: value.current.DebtLimit.value,
+        interestRate: value.current.InterestRate.value,
+        liquidationMargin: value.current.LiquidationMargin.value,
+        liquidationPadding: value.current.LiquidationPadding.value,
+        liquidationPenalty: value.current.LiquidationPenalty.value,
+        mintFee: value.current.MintFee.value,
+      });
+    }
+    return params;
+  };
+  const managerParamValues = await readManagerParams();
+
+  // upgrade the vaultFactory
+  const upgradeVaultFactory = async () => {
+    // @ts-expect-error cast XXX privateArgs missing from type
+    const { privateArgs } = kit;
+
+    const shortfallInvitation = await E(
+      E.get(reserveKit).creatorFacet,
+    ).makeShortfallReportingInvitation();
+
+    const poserInvitation = await E(
+      electorateCreatorFacet,
+    ).getPoserInvitation();
+    /** @type {import('../../src/vaultFactory/vaultFactory').VaultFactoryContract['privateArgs']} */
+    const newPrivateArgs = harden({
+      ...privateArgs,
+      // @ts-expect-error It has a value until reset after the upgrade
+      auctioneerInstance: auctioneerKit.instance,
+      initialPoserInvitation: poserInvitation,
+      initialShortfallInvitation: shortfallInvitation,
+      managerParams: managerParamValues,
+    });
+
+    const upgradeResult = await E(kit.adminFacet).upgradeContract(
+      bundleID,
+      newPrivateArgs,
+    );
+
+    console.log('upgraded vaultFactory.', upgradeResult);
+  };
+
+  // Wait for at least one new price feed to be ready before upgrading Vaults
+  void E.when(
+    any(
+      Object.values(vaultBrands).map(brand =>
+        E(priceAuthority).quoteGiven(AmountMath.make(brand, 10n), istBrand),
+      ),
+    ),
+    async price => {
+      trace(`upgrading after delay`, price);
+      await upgradeVaultFactory();
+      auctioneerKitProducer.reset();
+      // @ts-expect-error It has a value until reset just below
+      auctioneerKitProducer.resolve(auctioneerKit);
+      auctioneerProducer.reset();
+      // @ts-expect-error It has a value until reset just below
+      auctioneerProducer.resolve(auctioneerKit.instance);
+      // We wanted it to be valid for only a short while.
+      tempAuctioneerKit.reset();
+      await E(E(agoricNamesAdmin).lookupAdmin('instance')).update(
+        'auctioneer',
+        // @ts-expect-error It has a value until reset just below
+        auctioneerKit.instance,
+      );
+    },
+  );
+
+  console.log(`upgradeVaults scheduled; waiting for priceFeeds`);
+};
+
+const t = 'upgradeVaults';
```

Comment:
`t` in proposals is conventionally `true`. What's the difference again in using a string?

please giving it a distinct name to avoid confusing with `true`:
```suggestion
const uV = 'upgradeVaults';
```

Reply:
I think when it's a string, it can be used for drawing dependency graphs.

I saw this done recently, and I liked the pun with `t` and `true`, but I'll switch it. Puns should be avoided when they're confusing.